pkg/util/cluster/cluster.go

```diff
@@ -579,8 +584,10 @@ func (c *Cluster) createCluster(ctx context.Context, vnetResourceGroup, clusterN
 		if err != nil {
 			return err
 		}
-
-		oc.Properties.WorkerProfiles[0].VMSize = api.VMSizeStandardD2sV3
+		// In LocalDev mode, if workerVmSize is not default one, then it means user requested a specific one we need to keep.
```

Comment:
Nit: I'd rework this comment to explain what the conditional is doing rather than explaining what it's _not_ doing.

Reply:
I have a more concrete suggestion because it clicked after I walked away for a bit:

```suggestion
		// If we're in local dev mode and the user has not overridden the default VM size, use a smaller size for cost-saving purposes
```